web-app/js/portal/ui/openlayers/control/SpatialConstraint.js

```diff
@@ -13,7 +13,7 @@ Portal.ui.openlayers.SpatialConstraintType = {
 
```

Comment:
Good question.

It represents the control (a non-visible object) for the geometry you can draw on the map.
There is code to:
* Attach / remove it from the map
* Initialise its events
* Get the constraint / geometry that has been drawn
* Arrange the layer showing the geometry on the map
* Validating the geometry that is drawn
* Visualising an error if it occurs

So it does do a lot of things. I will have a think which of those responsibilities could be nicely extracted.